### PR TITLE
Add Nayeon by Abats Keyboard Club

### PR DESCRIPTION
**Pull request by ramonimbao**
## Description

Adds the Nayeon by Abats Keyboard Club.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/abatskeyboardclub/nayeon/config.h [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (drashna) [inline keyboards/abatskeyboardclub/nayeon/config.h:141]:
```suggestion
```

File: keyboards/abatskeyboardclub/nayeon/config.h [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+/*
+ * Split Keyboard specific options, make sure you have 'SPLIT_KEYBOARD = yes' in your rules.mk, and define SOFT_SERIAL_PIN.
+ */
+#define SOFT_SERIAL_PIN D0  // or D1, D2, D3, E6
+
+#define RGB_DI_PIN E2
+#ifdef RGB_DI_PIN
+#    define RGBLIGHT_LAYERS
+#    define RGBLIGHT_LAYERS_OVERRIDE_RGB_OFF
```
Comment (drashna) [inline keyboards/abatskeyboardclub/nayeon/config.h:58]: these should be defined at the keymap level, not the keyboard level. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/abatskeyboardclub/nayeon/config.h (hunk 2 above)
Comment (ramonimbao) [reply keyboards/abatskeyboardclub/nayeon/config.h]: Got it. I've moved the defines from the keyboard config to every keymap.

**Review by drashna — APPROVED** [review]

File: keyboards/abatskeyboardclub/nayeon/keymaps/default/config.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/default/config.h:16]:
```suggestion
#pragma once

```

File: keyboards/abatskeyboardclub/nayeon/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        for (uint8_t i=0; i<3; i++) {
+            rgblight_set_layer_state(i, false);
+        }
+        if (lock_bits < 3) {
+            rgblight_set_layer_state(lock_bits, true);
+        }
+    }
+
+    return res;
+}
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/default/keymap.c:66]:
```suggestion
bool led_update_user(led_t led_state) {
    uint8_t lock_bits = led_state.scroll_lock << 1 | led_state.caps_lock;
    for (uint8_t i=0; i<3; i++) {
        rgblight_set_layer_state(i, false);
    }
    if (lock_bits < 3) {
        rgblight_set_layer_state(lock_bits, true);
    }

    return false;
}
```

File: keyboards/abatskeyboardclub/nayeon/keymaps/iso/config.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/iso/config.h:16]:
```suggestion
#pragma once

```

File: keyboards/abatskeyboardclub/nayeon/keymaps/iso/keymap.c [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        for (uint8_t i=0; i<3; i++) {
+            rgblight_set_layer_state(i, false);
+        }
+        if (lock_bits < 3) {
+            rgblight_set_layer_state(lock_bits, true);
+        }
+    }
+
+    return res;
+}
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/iso/keymap.c:69]:
```suggestion
bool led_update_user(led_t led_state) {
    uint8_t lock_bits = led_state.scroll_lock << 1 | led_state.caps_lock;
    for (uint8_t i=0; i<3; i++) {
        rgblight_set_layer_state(i, false);
    }
    if (lock_bits < 3) {
        rgblight_set_layer_state(lock_bits, true);
    }
    return false;
}
```

File: keyboards/abatskeyboardclub/nayeon/keymaps/via/config.h [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/via/config.h:16]:
```suggestion
#pragma once

```

File: keyboards/abatskeyboardclub/nayeon/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        for (uint8_t i=0; i<3; i++) {
+            rgblight_set_layer_state(i, false);
+        }
+        if (lock_bits < 3) {
+            rgblight_set_layer_state(lock_bits, true);
+        }
+    }
+
+    return res;
+}
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/via/keymap.c:93]:
```suggestion
bool led_update_user(led_t led_state) {
    uint8_t lock_bits = led_state.scroll_lock << 1 | led_state.caps_lock;
    for (uint8_t i=0; i<3; i++) {
        rgblight_set_layer_state(i, false);
    }
    if (lock_bits < 3) {
        rgblight_set_layer_state(lock_bits, true);
    }
    return false;
}
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/abatskeyboardclub/nayeon/keymaps/iso/keymap.c [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    {0, 1, HSV_OFF}
+);
+
+const rgblight_segment_t* const PROGMEM rgb_layers[] = RGBLIGHT_LAYERS_LIST(ll_none, ll_cl, ll_sl);
+
+void keyboard_post_init_kb(void) {
+    rgblight_layers = rgb_layers;
+
+    keyboard_post_init_user();
+}
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/iso/keymap.c:53]:
```suggestion
void keyboard_post_init_user(void) {
    rgblight_layers = rgb_layers;
}
```

File: keyboards/abatskeyboardclub/nayeon/keymaps/iso/keymap.c [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+bool led_update_kb(led_t led_state) {
+    uint8_t lock_bits = led_state.scroll_lock << 1 | led_state.caps_lock;
+    for (uint8_t i=0; i<3; i++) {
+        rgblight_set_layer_state(i, false);
+    }
+    if (lock_bits < 3) {
+        rgblight_set_layer_state(lock_bits, true);
+    }
+}
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/iso/keymap.c:63]:
```suggestion
bool led_update_user(led_t led_state) {
    uint8_t lock_bits = led_state.scroll_lock << 1 | led_state.caps_lock;
    for (uint8_t i=0; i<3; i++) {
        rgblight_set_layer_state(i, false);
    }
    if (lock_bits < 3) {
        rgblight_set_layer_state(lock_bits, true);
    }
    return false;
}
```

File: keyboards/abatskeyboardclub/nayeon/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    {0, 1, HSV_OFF}
+);
+
+const rgblight_segment_t* const PROGMEM rgb_layers[] = RGBLIGHT_LAYERS_LIST(ll_none, ll_cl, ll_sl);
+
+void keyboard_post_init_kb(void) {
+    rgblight_layers = rgb_layers;
+
+    keyboard_post_init_user();
+}
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/default/keymap.c:50]:
```suggestion
void keyboard_post_init_user(void) {
    rgblight_layers = rgb_layers;
}
```

File: keyboards/abatskeyboardclub/nayeon/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+
+bool led_update_user(led_t led_state) {
+    uint8_t lock_bits = led_state.scroll_lock << 1 | led_state.caps_lock;
+    for (uint8_t i=0; i<3; i++) {
+        rgblight_set_layer_state(i, false);
+    }
+    if (lock_bits < 3) {
+        rgblight_set_layer_state(lock_bits, true);
+    }
+}
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/default/keymap.c:60]:
```suggestion
    return false;
}
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/abatskeyboardclub/nayeon/keymaps/iso/keymap.c [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    {0, 1, HSV_OFF}
+);
+
+const rgblight_segment_t* const PROGMEM rgb_layers[] = RGBLIGHT_LAYERS_LIST(ll_none, ll_cl, ll_sl);
+
+void keyboard_post_init_user(void) {
+    rgblight_layers = rgb_layers;
+}
+
+bool led_update_kb(led_t led_state) {
```
Comment (zvecr) [inline keyboards/abatskeyboardclub/nayeon/keymaps/iso/keymap.c:53]:
```suggestion
bool led_update_user(led_t led_state) {
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Please, just accept the suggestions via github. This is the 3rd time.

**Comment by ramonimbao** [comment]
Sorry. Thought it would be faster with just one commit.

**Review by fauxpark — APPROVED** [review]

File: keyboards/abatskeyboardclub/nayeon/config.h [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+#define RGB_DI_PIN E2
+#ifdef RGB_DI_PIN
+#    define RGBLED_NUM 26
+#    define RGBLIGHT_HUE_STEP 8
+#    define RGBLIGHT_SAT_STEP 8
+#    define RGBLIGHT_VAL_STEP 8
+//#    define RGBLIGHT_LIMIT_VAL 255 /* The maximum brightness level */
+//#    define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+/*== all animations enable ==*/
+//#    define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/abatskeyboardclub/nayeon/config.h:64]:
```suggestion
```

File: keyboards/abatskeyboardclub/nayeon/keymaps/iso/keymap.c [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                   KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      KC_RALT, KC_RGUI, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    )
+};
+
+#ifdef RGBLIGHT_ENABLE
+
+#define RGBLIGHT_LAYERS
+#define RGBLIGHT_LAYERS_OVERRIDE_RGB_OFF
+
```
Comment (drashna) [inline keyboards/abatskeyboardclub/nayeon/keymaps/iso/keymap.c:33]:
```suggestion
```

File: keyboards/abatskeyboardclub/nayeon/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
+        _______, _______, _______,                            _______,                                     _______, _______, _______,          _______, _______, _______
+    )
+};
+
+#ifdef RGBLIGHT_ENABLE
+
+#define RGBLIGHT_LAYERS
+#define RGBLIGHT_LAYERS_OVERRIDE_RGB_OFF
+
```
Comment (drashna) [inline keyboards/abatskeyboardclub/nayeon/keymaps/via/keymap.c:57]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**